lib/src/main/java/com/grio/lib/features/editor/views/ToolOptions.kt

```diff
@@ -126,20 +59,41 @@ class ToolOptions @JvmOverloads constructor(
         return super.onTouchEvent(event)
     }
 
+    /**
+     * Sets the tool size for tool options
+     *
+     * @param percentage the size to set the tool
+     */
+    fun setToolSize(percentage: Int) {
+        size_slider.progress = percentage
+    }
+
     /**
      * Sets the drawer to show options based on the selected tool
      *
-     * @param type the type of drawer format to display
+     * @param tool the type of drawer format to display
      */
-    fun setDrawerType(type: ToolDrawerType) {
-        toolOptionsDrawerType = type
-        if (toolOptionsDrawerType == ToolDrawerType.SLIDER) {
-            slider.visibility = View.VISIBLE
-            shapeButtonHolder.visibility = View.GONE
-        } else if (toolOptionsDrawerType == ToolDrawerType.SHAPES) {
-            slider.visibility = View.GONE
-            shapeButtonHolder.visibility = View.VISIBLE
+    fun setDrawerType(tool: Tool) {
+        currentTool = tool
+        color_label.visibility = View.VISIBLE
+        color_picker.visibility = View.VISIBLE
+        if (currentTool == Tool.PEN || currentTool == Tool.TEXT) {
+            shape_button_holder.visibility = View.GONE
+            shape_label.visibility = View.GONE
+            size_slider.visibility = View.VISIBLE
```

Comment:
Some of these `View`s can likely be placed in a `Group` so that their visibility can be toggled at the same time.

Reply:
tool options view is currently implemented using constraint layout, adding those views to groups would save a line or two here but require inserting multiple view groups and refactoring the layout to accommodate them. Going to leave as is for now and possibly revisit at a later date if it makes sense with a new design